Migration of image optimization content (#3269)
GoogleChrome/web.dev: src/site/content/en/fast/choose-the-right-image-format/index.md

<figure class="w-figure">
  <img src="./vector-raster-zoom.png" alt="Zoomed-in vector and raster images">
  <figcaption>Zoomed-in vector image (L) raster image (R)</figcaption>
</figure>

* [Vector graphics](https://en.wikipedia.org/wiki/Vector_graphics)
use lines, points, and polygons to represent an image.
* [Raster graphics](https://en.wikipedia.org/wiki/Raster_graphics)
represent an image by encoding the individual values of each pixel within a rectangular grid.

Each format has its own set of pros and cons.
Vector formats are ideally suited for images that consist of simple geometric shapes such as logos, text, or icons.
They deliver sharp results at every resolution and zoom setting,
which makes them an ideal format for high-resolution screens and assets that need to be displayed at varying sizes.

However, vector formats fall short when the scene is complicated (for example, a photo):
the amount of SVG markup to describe all the shapes can be prohibitively high
and the output may still not look "photorealistic".
When that's the case, that's when you should be using a raster image format

> such as GIF, PNG, JPEG, or WebP.

Reviewer: ```suggestion
such as PNG, JPEG, or WebP.
```

the total number of pixels increases by a factor of four:
double the number of horizontal pixels, times double the number of vertical pixels.
Hence, a "2x" screen not just doubles, but quadruples the number of required pixels!

So, what does this mean in practice?
High-resolution screens enable you to deliver beautiful images, which can be a great product feature.
However, high-resolution screens also require high-resolution images, therefore:

* Prefer vector images whenever possible as they are resolution-independent and always deliver sharp results.
* If a raster image is required, serve [responsive images](/serve-responsive-images/).

## Features of different raster image formats

In addition to different lossy and lossless compression algorithms,
different image formats support different features such as animation and transparency (alpha) channels.
As a result, the choice of the "right format" for a particular image is a combination of desired visual results and functional requirements.

<div class="w-table-wrapper">
<table>
<thead>
  <tr>
    <th>Format</th>
    <th>Transparency</th>
    <th>Animation</th>
    <th>Browser</th>
  </tr>
</thead>
<tbody>

> <tr>
>   <td data-th="format"><a href="http://en.wikipedia.org/wiki/Graphics_Interchange_Format">GIF</a></td>
>   <td data-th="transparency">Yes</td>
>   <td data-th="animation">Yes</td>
>   <td data-th="browser">All</td>
> </tr>

Reviewer: ```suggestion
```

<tr>
  <td data-th="format"><a href="http://en.wikipedia.org/wiki/Portable_Network_Graphics">PNG</a></td>
  <td data-th="transparency">Yes</td>
  <td data-th="animation">No</td>
  <td data-th="browser">All</td>
</tr>
<tr>
  <td data-th="format"><a href="http://en.wikipedia.org/wiki/JPEG">JPEG</a></td>
  <td data-th="transparency">No</td>
  <td data-th="animation">No</td>
  <td data-th="browser">All</td>
</tr>
<tr>
  <td data-th="format"><a href="http://en.wikipedia.org/wiki/WebP">WebP</a></td>
  <td data-th="transparency">Yes</td>
  <td data-th="animation">Yes</td>
  <td data-th="browser">All modern browsers. See <a href="https://caniuse.com/#feat=webp">Can I use?</a></td>
</tr>
</tbody>
</table>
</div>

> There are three universally supported raster image formats: GIF, PNG, and JPEG.

Reviewer: ```suggestion
There are two universally supported raster image formats: PNG, and JPEG.
```

In addition to these formats, modern browsers support the newer format WebP,
which offers better overall compression and more features. So, which format should you use?

The WebP format will generally provide better compression than older formats,
and should be used where possible.
You can use WebP along with another image format as a fallback.
See [Use WebP images](/serve-images-webp/) for more details.

In terms of older image formats, consider the following:

> 1. **Do you need animation? Use GIF.**
>     * GIF limits the color palette to at most 256 colors, which makes it a poor choice for most images.
>     Further, PNG-8 delivers better compression for images with a small palette.
>     As a result, GIF is the right answer only when animation is required.

Reviewer: ```suggestion
1. **Do you need animation? Use `<video>` elements.**
    * What about GIF? GIF limits the color palette to at most 256 colors,
      and creates significantly larger file sizes than videos. See
      [Replace animated GIFs with video](/replace-gifs-with-videos/).
```

1. **Do you need to preserve fine detail with highest resolution? Use PNG.**
    * PNG does not apply any lossy compression algorithms beyond the choice of the size of the color palette.
    As a result, it will produce the highest quality image,
    but at a cost of significantly higher filesize than other formats. Use judiciously.
    * If the image asset contains imagery composed of geometric shapes, consider converting it to a vector (SVG) format!
    * If the image asset contains text, stop and reconsider. Text in images is not selectable, searchable, or "zoomable".
    If you need to convey a custom look (for branding or other reasons), use a web font instead.
1. **Are you optimizing a photo, screenshot, or a similar image asset? Use JPEG.**

>     * JPEG uses a combination of lossy and lossless optimization to reduce filesize of the image asset. Try several JPEG quality levels to find the best quality vs. filesize tradeoff for your asset.

Reviewer 2: ```suggestion
    * JPEG uses a combination of lossy and lossless optimization to reduce filesize of the image asset. Try several JPEG quality levels to find the best quality versus filesize tradeoff for your asset.
```

  <td data-th="format"><a href="http://en.wikipedia.org/wiki/WebP">WebP</a></td>
  <td data-th="transparency">Yes</td>
  <td data-th="animation">Yes</td>

>   <td data-th="browser">Chrome, Opera, Android, Safari (from version 14)</td>

Reviewer 2: It's also supported in Firefox since version 65 (early 2019). Not sure what's the best way to phrase this...

Author: maybe we should phrase it at this point as "all modern browsers" and then link that to MDN/Can I Use?

Author: I completely destroyed my local branch trying to rebase it so I'll need to start over with it to make the change.

Author: ok I've done this.

If I rebase, I will end up with all of the individual commits listed like last time, which was not what was wanted. Is there something I should be doing to stop that happening?

Reviewer 2: I don't think the commit history matters much if we do squash and merge, paging @robdodson for advice :) 

Context: This branch has the outdated version of the /fast page and I suggested rebasing. The lazy-loading section has been added in the meantime and lazysizes post has been moved there. (I suggested rebasing to make sure we don't end up with duplicated lazysizes post because I'm not sure if this would show up as a conflict when you attempt merging.)

Reviewer 3: > I don't think the commit history matters much if we do squash and merge, paging @robdodson for advice :)

Yeah if you're doing a squash and merge from a PR branch then all of the commits will get combined into one.

> (I suggested rebasing to make sure we don't end up with duplicated lazysizes post because I'm not sure if this would show up as a conflict when you attempt merging.)

That's a good question. One option would be to manually do a squash merge to see what happens. I think the commands would be something like this:

```
git checkout -b rachelandrew-image-optimize master
git pull https://github.com/rachelandrew/web.dev.git image-optimize
git merge --squash rachelandrew-image-optimize
```
